### PR TITLE
[dv] Split sec_cm_testplan into multiple testplans

### PR DESCRIPTION
**Pull request by weicaiyang**
Addressed Sri's comment in #8481
Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  testpoints: [
+    {
+      name: hardened_counter_check
+      desc: ''' Verify design behavior is correct when the harden counter is changed to an
+            unexpected value.
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson:9]: Nit keep this in single line. 
```
Verify that violating prim_count counter properties generate an alert.
```

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  testpoints: [
+    {
+      name: hardened_counter_check
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson:7]: Will be nice to have the name of the prim module reflected in the name. Also, I thought we were following some conventions in the vseq too right? We should match everything up.

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      desc: ''' Verify design behavior is correct when the harden counter is changed to an
+            unexpected value.
+
+            Stimulus:
+            - At the falling edge (non-active edge), force the counter to a different value.
+            - Randomly flip the value back to any other value to ensure the error is latched and
+              won't go away until reset.
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson:14]:
```suggestion
            Stimulus:
            - Randomly pick one of the two counters. 
            - At the falling edge (non-active edge), force-deposit the counter to a different value than expected.
            - On the next cycle, the violation of hardened counter property should generate an alert.
            - Repeat for ALL prim_count instance in the DUT. 
```

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+            Same checks as `one_hot_check`'''
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson:16]: Perhaps copy-paste, since it is in a different file (which may or may not be imported)?
Also, put the closing `'''` on newline.

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 5]
```diff
@@ -179,12 +180,21 @@
...
+    }
+    {
+      name: sec_cm_additional_check
       desc: '''
-            Verify the outcome after injecting faults into security countermeasures.
+            Verify the outcome after injecting faults to security countermeasures.
 
             Stimulus:
-            As mentioned in `sec_cm_testplan`.
+            As mentioned in `sec_cm_count_testplan` and `sec_cm_fsm_testplan`.
```
Comment (sriyerg) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:197]: In the expanded testplan, all testpoints show up in one table. the actual testplan is not reflected anywhere. We should perhaps reword this to make it clearer.

File: hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson [hunk 6]
```diff
@@ -0,0 +1,21 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  testpoints: [
+    {
+      name: redundant_encoding_fsm_check
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson:7]: Same comments as above.

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson (hunk 2 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson]: sure, updated to prim_count_check. vseq uses prim_count.

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson (hunk 1 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson]: Updated.

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson (hunk 3 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson]: I actually don't randomly pick the counters. just always use one counter, which makes the sequence less complex. 
Also alert may not arrive right on the next cycle, as alert is async
Updated the stimulus.

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson (hunk 4 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson]: make sense. Updated, thanks

File: hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson (hunk 6 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson]: done

File: hw/ip/keymgr/data/keymgr_testplan.hjson (hunk 5 above)
Comment (weicaiyang) [reply hw/ip/keymgr/data/keymgr_testplan.hjson]: changed to prim_*_check. Thanks

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    {
+      name: prim_count_check
+      desc: ''' Verify that violating prim_count counter properties generate a fatal alert.
+
+            Stimulus:
+            - At the falling edge (non-active edge), force the counter to a different value than
+              expected.
+            - Randomly flip the value back to any other value to ensure the error is latched and
+              won't go away until reset.
+            - In the next a few cycle, the violation of hardened counter property should generate a
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson:15]:
```suggestion
            - Within the few cycles, the violation of hardened counter property should generate a
```

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+              fatal alert.
+            - Repeat for ALL prim_count instance in the DUT.
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson:17]:
```suggestion
            - Repeat for ALL prim_count instances in the DUT.
```

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+{
+  testpoints: [
+    {
+      name: prim_count_check
+      desc: ''' Verify that violating prim_count counter properties generate a fatal alert.
+
+            Stimulus:
+            - At the falling edge (non-active edge), force the counter to a different value than
+              expected.
+            - Randomly flip the value back to any other value to ensure the error is latched and
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson:13]: Not sure about this step - it it different from the first?

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+              won't go away until reset.
+            - In the next a few cycle, the violation of hardened counter property should generate a
+              fatal alert.
+            - Repeat for ALL prim_count instance in the DUT.
+
+            Checks:
+            - Check that fatal alert is triggered.
+            - Check that err_code/fault_status is updated correctly and preserved until reset.
+            - Check the following operation should be failed if applicable.
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson:22]: Perhaps clarify what "following operation" means? 
```suggestion
            - Verify any operations that follow fail (as applicable).
```

File: hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+// SPDX-License-Identifier: Apache-2.0
+{
+  testpoints: [
+    {
+      name: prim_fsm_check
+      desc: ''' Verify that entering to an undefined state generates a fatal alert.
+
+            Stimulus:
+            - Backdoor force the FSM to any of the undefined values.
+            - Randomly flip the value back to a defined state to ensure the error is latched and
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson:12]: Is this step needed? I thought a value different from the known states would land it in the default case triggering an alert.

File: hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+               won't go away until reset.
+            - In the next a few cycle, the violation of hardened FSM property should generate a
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson:14]:
```suggestion
            - Within the next few cycles, the FSM landing in an invalid state should trigger a
```

File: hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+              fatal alert.
+            - Repeat for ALL prim_count instance in the DUT.
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson:16]:
```suggestion
            - Repeat for ALL such hardened FSMs in the DUT.
```

File: hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+      name: prim_one_hot_check
+      desc: ''' Verify that violating one-hot property generates a fatal alert.
+
+            Stimulus:
+            - Backdoor force one-hot coding signals to not one-hot.
+            - Randomly flip the value back to ensure the error is latched and won't go away until
+              reset.
+            - In the next a few cycle, the violation of one-hot property should generate a fatal
+              alert.
+            - Repeat for ALL prim_count instance in the DUT.
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson:16]: Same comments as above.

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 15]
```diff
@@ -179,12 +180,21 @@
       tests: ["keymgr_sync_async_fault_cross"]
     }
     {
-      name: sec_cm_check
+      name: stress_all
+      desc: '''
+            - Combine above sequences in one test to run sequentially, except csr sequence and
+              keymgr_cfg_regwen (requires zero_delays).
+            - Randomly add reset between each sequence.'''
```
Comment (sriyerg) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:187]:
```suggestion
            - Randomly add reset between each sequence.
            '''
```

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 16]
```diff
@@ -179,12 +180,21 @@
...
+      milestone: V2
+      tests: ["keymgr_stress_all"]
+    }
+    {
+      name: sec_cm_additional_check
       desc: '''
-            Verify the outcome after injecting faults into security countermeasures.
+            Verify the outcome after injecting faults to security countermeasures.
```
Comment (sriyerg) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:194]:
```suggestion
            Verify the outcome of injecting faults to security countermeasures.
```

File: hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson (hunk 9 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson]: This is to flip the counter to normal value. Reworded. LMK If it's still not clear.
Step 2 of the stimulus at [slides](https://docs.google.com/presentation/d/1hcS2hipBXW4TmIna8xfn83mfk2xIMekxE2xERhD_i38/edit#slide=id.gf56c479c22_0_5)

File: hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  testpoints: [
+    {
+      name: prim_one_hot_check
+      desc: ''' Verify that violating one-hot property generates a fatal alert.
+
+            Stimulus:
+            - Backdoor force one-hot coding signals to not one-hot.
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson:11]:
```suggestion
            - Backdoor force the one-hot encoded signals with multiple bits set.
```

File: hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson [hunk 18]
```diff
@@ -0,0 +1,28 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  testpoints: [
+    {
+      name: prim_one_hot_check
+      desc: ''' Verify that violating one-hot property generates a fatal alert.
+
+            Stimulus:
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson:10]: Add a line above Stimulus to make it clear:
```
      One-hot encoded signals are captured in `prim_one_hot` module. 
```

Otherwise the stimulus suggests forcing one-hot encoded signals directly (which would then invite more questions).

File: hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson (hunk 17 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson]: Updated. Thanks

File: hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson (hunk 18 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/testplans/sec_cm_one_hot_testplan.hjson:11]: Added, thanks

**Review by sriyerg — APPROVED** [review]
